client/src/App.jsx

```diff
@@ -251,6 +272,34 @@ class App extends React.Component {
     return new Date(this.state.checkOut);
   }
 
+  showGuestPicker() {
```

Comment:
Your showGuestPicker() method theoretically could hide the guest picker since it negates the current state. Perhaps this could be called "toggleGuestPicker" if it should be able to do both, or only set the state to `true` if its purpose is to show it.